### DISCUSSION TITLE
Vercel Successful build but I get blank page and these errors

### DISCUSSION THREAD

**Discussion by mhesham32** [vercel/next.js#20455] [Help] [answered]
hello I'm working on this next.js project and the only thing I have updated was adding getInitialProps on _app.js but this shouldn't be the problem I'm not sure though this is the code for it 

```js
MyApp.getInitialProps = async (appContext) => {
  const { res, pathname, query } = appContext.ctx;
  const sendRedirectLocation = (location = '/') => {
    if (pathname !== '/') {
      res.writeHead(302, {
        Location: location,
      });
      res.end();
      return { props: {} }; // stop execution
    }
  };
  try {
    const authToken = nextCookie(appContext.ctx).authToken || null;

    if (authToken) {
      const verifyUserToken = functions.httpsCallable('verifyUserToken');
      const userRes = await verifyUserToken({ authToken });
      const { role, experienceId } = userRes.data?.customClaims || {};

      console.log({
        role,
        experienceId,
        isAllowedRoute: isAllowedRoute(query, pathname, role, experienceId),
      });
      if (!isAllowedRoute(query, pathname, role, experienceId)) {
        sendRedirectLocation();
      }
    } else if (!isAllowedRoute(query, pathname, null, null)) {
      sendRedirectLocation();
    }

    return { props: {} };
  } catch (error) {
    if (!isAllowedRoute(query, pathname, null, null)) {
      sendRedirectLocation();
    }
    console.error(error);
    return { props: {} };
  }
};
```
after implementing that I build the project on my dev and it worked without any problems and when I published it to vercel it was built successfully but showing the blank page on the preview URL and these errors on the console

GET https://domain-3f8hep6kz.vercel.app/%PUBLIC_URL%/favicon.ico 400
===
GET https://domain-3f8hep6kz.vercel.app/%PUBLIC_URL%/manifest.json 400
===
Manifest: Line: 1, column: 1, Syntax error.
===
The attempt to bind "/%PUBLIC_URL%/manifest.json" in the workspace failed as this URI is malformed.
===

**Comment by mhesham32** [comment, marked as answer]
I think the problem was with the cache when the build didn't use the cached data it was solved

**Reply by benjaminben** [reply]
Hey there, I'm having a similar issue, home page of the site is blank (only in production) with the following error in the console `https://<siteurl>/%PUBLIC_URL%/favicon.ico Failed to load resource: the server responded with a status of 400 ()` Can you please give more detail on how you resolved this?

**Reply by benjaminben** [reply]
Resolved by upgrading Next 😅 sigh
